### PR TITLE
Manas week 2

### PR DESCRIPTION
**Pull request by Manas23601**
Created an empty file for knapsack_0_1

### CODE REVIEW

**Comment by Manas23601** [comment]
How do I stop Github actions from running build each and every time I push some changes

**Comment by Manas23601** [comment]
I think I have created a basic skeleton(still incomplete) for now.  If there are problems, please let me know. Please direct me to any resources if necessary

**Comment by krashish8** [comment]
> How do I stop Github actions from running build each and every time I push some changes?

I don't think you should do that. It's good to see the Actions result after every push. Initially, they are going to fail.
Anyway, if you want, you can temporarily disable the corresponding actions in the Actions tab **for your fork**, or can comment the corresponding workflow file.

**Comment by krashish8** [comment]
@Manas23601 Maybe it's a good idea to reconsider the naming of the functions or the directory structure.
You can probably ask in the pgRouting Gitter channel for the same.

I think you should only use underscores in the function name, not dashes. Some function names suggestions:
- vrp_or_knapsack_0_1
- vrp_or_knapsack
- vrp_knapsack_0_1 (OR stands for Operations Research tools, but it may mislead the users sometimes)
- vrp_knapsack
- vrp_knapsack_01

The directory could be `knapsack` only, or `or_tools` or something related, but not `knapsack_0-1`.

File: sql/knapsack_0-1/_knapsack_0-1.sql [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+ ********************************************************************PGR-GNU*/
+
+
+CREATE OR REPLACE FUNCTION _vrp_or_knapsack_0-1(
```
Comment (cvvergara) [inline sql/knapsack_0-1/_knapsack_0-1.sql]: postgres names can not have dashes.

File: sql/knapsack_0-1/_knapsack_0-1.sql [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/*PGR-GNU*****************************************************************
+File: _knapsack_0-1.sql
+
+Copyright (c) 2021 pgRouting developers
+Mail: project@pgrouting.org
+
+Function's developer:
+Copyright (c) 2021 Celia Virginia Vergara Castillo
```
Comment (cvvergara) [inline sql/knapsack_0-1/_knapsack_0-1.sql]: I am not developing this function.
Remove my name

File: sql/knapsack_0-1/knapsack_0-1.sql [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+ ********************************************************************PGR-GNU*/
+
+
+CREATE OR REPLACE FUNCTION _vrp_or_knapsack_0-1(
```
Comment (cvvergara) [inline sql/knapsack_0-1/knapsack_0-1.sql]: Postgres names can not have dashes.
(I really dont know of a language that can have dashes on the names, as a dash is the minus sign)

File: sql/knapsack_0-1/knapsack_0-1.sql [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/*PGR-GNU*****************************************************************
+File: _knapsack_0-1.sql
+
+Copyright (c) 2021 pgRouting developers
+Mail: project@pgrouting.org
+
+Function's developer:
+Copyright (c) 2021 Celia Virginia Vergara Castillo
```
Comment (cvvergara) [inline sql/knapsack_0-1/knapsack_0-1.sql]: Remove my name

File: sql/knapsack_0-1/_knapsack_0-1.sql [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+Copyright (c) 2021 Joseph Emile Honour Percival
```
Comment (cvvergara) [inline sql/knapsack_0-1/_knapsack_0-1.sql]: I don't think Iosefa is developing this function.
Remove his name

File: sql/knapsack_0-1/_knapsack_0-1.sql [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
```
Comment (cvvergara) [inline sql/knapsack_0-1/_knapsack_0-1.sql]: I do not see you name as the function developer.
Ar you going to give Iosefa and me the credit for your work?

File: sql/knapsack_0-1/knapsack_0-1.sql [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+Copyright (c) 2021 Joseph Emile Honour Percival
+
```
Comment (cvvergara) [inline sql/knapsack_0-1/knapsack_0-1.sql]: I do not see you name as the function developer.
Ar you going to give Iosefa and me the credit for your work?

File: sql/knapsack_0-1/knapsack_0-1.sql [hunk 8]
```diff
@@ -0,0 +1,51 @@
+/*PGR-GNU*****************************************************************
+File: _knapsack_0-1.sql
+
+Copyright (c) 2021 pgRouting developers
+Mail: project@pgrouting.org
+
+Function's developer:
+Copyright (c) 2021 Celia Virginia Vergara Castillo
+Copyright (c) 2021 Joseph Emile Honour Percival
```
Comment (cvvergara) [inline sql/knapsack_0-1/knapsack_0-1.sql]: Remove Iosefa's name

File: sql/knapsack_0-1/knapsack_0-1.sql [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+RETURNS SETOF RECORD AS
+$BODY$
+    SELECT *
+    FROM _vrp_or_knapsack_0-1(_pgr_get_statement($1), _pgr_get_statement($2), $3);
+$BODY$
+LANGUAGE SQL VOLATILE STRICT;
+
+-- COMMENTS
+
+COMMENT ON FUNCTION __vrp_or_knapsack_0-1(TEXT, TEXT, INTEGER)
```
Comment (cvvergara) [inline sql/knapsack_0-1/knapsack_0-1.sql]: I am no compiler, but the first underscore looks way too long.

**Review by cvvergara — CHANGES_REQUESTED** [review]

**Comment by cvvergara** [comment]
I am wondering, is there a knapsack that is not 0-1?
For the moment in vrpRouting the answer is NO.
`vrp_knapsack` I prefer as the name, and if in there future there is another kind of knapsack we will figure out the name

**Comment by Manas23601** [comment]
I will immediately look at the name changes. I think I forgot to commit those changes

**Pull request merged**